### PR TITLE
Initial flask setup

### LINKED ISSUE #21: General Flask backend Setup + OpenAPI REST Documentation
We need to set up a boilerplate for our flask server with the right starting tools to add endpoints. This is for the backend and should include the basic react app, and any frameworks that will support an easy handoff to the front end.

### PR DESCRIPTION
**Pull request by UTkzhang**
Initial set up for flask backend, including models, sqlite db, api with openapi documentation (REST), mailing service, and docker.

### CODE REVIEW

File: PlateUp-flask/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,6 @@
+FROM python:3.8
+COPY . /app
+WORKDIR /app
+RUN pip install -r requirements.txt
+ENTRYPOINT ["python"]
```
Comment (eanile) [inline PlateUp-flask/Dockerfile]: Should we specify python3 just in case?

File: .gitignore [hunk 2]
```diff
@@ -4,3 +4,6 @@ PlateUp/.expo-shared
 
 # Node
 PlateUp/node_modules
+
+# Flask
+PlateUp-flask/email_config.ini
```
Comment (eanile) [inline .gitignore:9]: Is there a reason this is ignored?

File: PlateUp-flask/Dockerfile (hunk 1 above)
Comment (UTkzhang) [reply PlateUp-flask/Dockerfile]: sure

File: .gitignore (hunk 2 above)
Comment (UTkzhang) [reply .gitignore:9]: email password/user, should be configured locally

File: PlateUp-flask/emailservice.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import configparser
+import smtplib
+from email.mime.text import MIMEText
```
Comment (eanile) [inline PlateUp-flask/emailservice.py:4]: nit: try and keep the format of the imports the same in each py file. Example: all "import ..." statements before "from __ import ..." statements, and keep the statements within each group in alphabetical order.

Comment (UTkzhang) [reply PlateUp-flask/emailservice.py:4]: fixed

File: PlateUp-flask/run.py [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+        if user is not None and check_password_hash(user.password, password):
+            login_user(user)
+            return Response("Login successful. User %s"% userId, status=200)
+
+        return Response("403 Forbidden", status=403)
+
+    @loginR.doc(description="Logging current user out.")
+    def delete(self):
+        logout_user(current_user)
+        return Response("Login successful. User %s"%current_user.id, status=200)
```
Comment (eanile) [inline PlateUp-flask/run.py]: Login successful -> Logout successful

File: PlateUp-flask/run.py [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+
+
+# Mail API
+@mailR.route('')
+class Mail(Resource):
+    # @login_required
+    @mailR.doc(description="Sends a welcome email to user with their client ID and default password information (Development Email Only).")
```
Comment (eanile) [inline PlateUp-flask/run.py:122]: nit: line too long

File: PlateUp-flask/run.py [hunk 6]
```diff
@@ -0,0 +1,203 @@
...
+
+# -----------------------------------------------------------------------------
+# Background tasks
+# -----------------------------------------------------------------------------
+# update stuff as a scheduled job while server is active
+@scheduler.scheduled_job('interval', seconds=3600)
+def updateRecipes():
+    print("updating recipes...")
+    # do some process
+    time.sleep(10)
```
Comment (eanile) [inline PlateUp-flask/run.py:193]: I'm assuming this is temporary?

File: PlateUp-flask/models.py [hunk 7]
```diff
@@ -1,9 +1,10 @@
+import json
```
Comment (eanile) [inline PlateUp-flask/models.py]: Still not in alpha order. Also other files have this issue. Not a huge thing, just something I noticed.

File: PlateUp-flask/run.py (hunk 4 above)
Comment (eanile) [reply PlateUp-flask/run.py]: Done.

File: PlateUp-flask/models.py (hunk 7 above)
Comment (eanile) [reply PlateUp-flask/models.py]: Fixed.

**Comment by eanile** [comment]
NOTE: also please change class names of APIs to *API. i.e. Login -> LoginAPI.

**Pull request merged**